### PR TITLE
feat: add gitee provider

### PR DESCRIPTION
**Pull request by sh1luo**
Signed-off-by: wasabi <690898835@qq.com>

### CODE REVIEW

**Comment by sh1luo** [comment]
demo: http://test.rescld.cn/login/oauth/authorize?client_id=b4690f977fb0e2c609bf&response_type=code&redirect_uri=http://test.rescld.cn/callback&scope=read&state=casdoor

click gitee logo, use your gitee account to login,

Next, open manager page: http://test.leviatan.cn/
account: admin
password: 123

You can see that the User table adds a record about you.

File: idp/gitee.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+		AccessToken:  tokenResp.AccessToken,
+		TokenType:    tokenResp.TokenType,
+		RefreshToken: tokenResp.RefreshToken,
+		Expiry:       time.Unix(time.Now().Unix()+int64(tokenResp.ExpiresIn), 0),
+	}
+
+	return token, nil
+}
+
+type GiteeUserResponse struct {
```
Comment (hsluoyz) [inline idp/gitee.go:116]: provide json example like other cases

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Review by hsluoyz — APPROVED** [review]

**Comment by hsluoyz** [comment]
@sh1luo 

image

**Comment by sh1luo** [comment]
After the branch was merged in, I shut down the server, wait a moment, I opened it again

**Comment by sh1luo** [comment]
can visit now.

**Comment by hsluoyz** [comment]
@sh1luo did you support the other mode? As example: https://forum.casbin.com/ , click login:

image

It should show a big button like this.

**Pull request merged**